src/unittest/test_irr_gltf_mesh_loader.cpp

```diff
@@ -0,0 +1,368 @@
+// Minetest
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#include "CSceneManager.h"
+#include "content/subgames.h"
+#include "filesys.h"
+
+#include "CReadFile.h"
+#include "vector3d.h"
+
+#include <irrlicht.h>
+
+#include "catch.h"
+
+using v3f = irr::core::vector3df;
+using v2f = irr::core::vector2df;
```

Comment:
For two aliases?

Reply:
And for including vector3d.
I mean, all other files do it like this.
You can also include `"irr_v3d.h"` and `"irr_v2d.h"`.

Reply:
> And for including vector3d. I mean, all other files do it like this.

I looked at it and I don't think this would be good practice: It includes AABBs and `SColor`, both of which aren't needed here. The name (and the corresponding collection of types) is also somewhat vague. Why are AABBs included for example, but matrices and quaternions are not? I can't think of a simple rule these types follow. To me it looks like it's just a relatively arbitrary mashup of 4 commonly used types that should probably be refactored away.

> You can also include `"irr_v3d.h"` and `"irr_v2d.h"`.

Thanks, these seem like a better fit. Done.